errorcodes.md

```diff
@@ -0,0 +1,25 @@
+# API Error codes
+
+This document shows the list of error codes that can be returned from the API, along with their HTTP Codes.
+
+
+| Name | Code | Description |
+|:---------------- |:--------------|:----------------------|
+|MISSING_HEADERS|400|Some headers were missing from the request: %s|
+|USER_NOT_FOUND|400|User credentials not found|
+|TOKEN_DOES_NOT_MATCH_SERVICE|401|The provided token does not belong to the provided service.|
+|ACTION_NOT_FOUND|400|The system wasn't able to guess the action type or the request|
+|WRONG_XML_PAYLOAD|400|The system wasn't able to parse the given XML payload|
+|WRONG_JSON_PAYLOAD|400|The system wasn't able to parse the given JSON payload (either it was empty or wrong)|
+|WRONG_XML_VALIDATION_RESPONSE|500|The system wasn't able to parse the XML obtained from the Access Control|
+|KEYSTONE_CONNECTION_ERROR|500|There was a connection error accessing the Access Control: %s|
+|KEYSTONE_VALIDATION_ERROR|500|The Access Control failed to make a decision due to the following error: %s|
+|KEYSTONE_AUTHENTICATION_ERROR|500|There was a connection error while authenticating to Keystone: %s|
+|KEYSTONE_AUTHENTICATION_REJECTED|401|User authentication was rejected with code: %s|
+|KEYSTONE_SUBSERVICE_NOT_FOUND|401|Could not find subservice with name [%s] in Keystone.|
+|PEP_PROXY_AUTHENTICATION_REJECTED|500|Proxy authentication was rejected with code: %s|
+|ROLES_NOT_FOUND|401|No roles were found for the user token in the give subservice: %s|
+|ACCESS_DENIED|403|The user does not have the appropriate permissions to access the selected action|
+|TEMPLATE_LOADING_ERROR|500|There was an error loading the templates for the validation Request: %s|
+|UNEXPECTED_CONTENT_TYPE|400|The MIME content type received is not supported: %s|
```

Comment:
I think 416 is the proper one for this... Or are you using 400 intentionally to keep the number of different codes small enought?